src/boot/efi/vmm.c

```diff
@@ -319,6 +339,54 @@ const char* smbios_find_oem_string(const char *name) {
         return NULL;
 }
 
+static const char* smbios_get_string(const SmbiosHeader *header, uint16_t string, uint64_t left) {
```

Comment:
a parameter called "string" that is an integer is a bit weird. maybe call this nr? or "needle"? (in the sense of needle/haystack)?

Reply:
hmm, afaics all values you actually pass as 2nd argument are of type uint8_t? it seems surprising to use uint16_t here then?

that said, I think we should change this to size_t. It's how we generally index and size arbitrary memory, entirely generically.

The third arg should also be size_t really